### PR TITLE
Fold InitialSearchPhase into AbstractSearchAsyncAction

### PR DESCRIPTION
**Pull request by javanna**
Historically, we have two base classes for search actions that generally need to fan out to multiple shards and then move on to the following phase: InitialSearchPhase and AbstractSearchAsyncAction that extends it. Practically, every search action extends the latter, and there are no direct subclasses of InitialSearchPhase in our codebase.

This commit folds InitialSearchPhase into AbstractSearchAsyncAction in the attempt of simplifying things and making the search code running on the coordinating node easier to reason about.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java [hunk 1]
```diff
@@ -213,35 +209,13 @@ public void sendCanMatch(
...
                 null,
-                (iter) -> new InitialSearchPhase<SearchPhaseResult>("test", searchRequest,
-                        iter, logger, randomIntBetween(1, 32), executor) {
+                iter -> new SearchPhase("next") {
                     @Override
-                    void onPhaseDone() {
+                    public void run() {
                         latch.countDown();
                     }
-
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java]: I am curious to hear if I am over-simplifying things here. This is how the next phase used to be before #27069 . I wonder if it's necessary for the next phase to be an AbstractSearchAsyncAction subclass at this point, which is not so easy to recreate.

**Comment by javanna** [comment]
run elasticsearch-ci/2

**Review by jimczi — APPROVED** [review]
Nice cleanup, I left one comment regarding a test but lgtm otherwise.

**Comment by jimczi** [comment]
@elasticmachine update branch

**Pull request merged**